### PR TITLE
Remove GDExtension compatibility code for Godot 4.0

### PR DESCRIPTION
**Pull request by dsnopek**
Now that PR #77410 has been merged, there is no way to allow GDExtensions built for Godot 4.0 to work with Godot 4.1.

So, this PR remove all the compatibility code that allowed that to work, so that Godot 4.1+ will simply refuse to load a GDExtension made for Godot 4.0 with a nice message (as opposed to crashing :-)).

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]
Makes sense to me. I believe this was also the consensus when discussed at the Godot Sprint.

File: core/extension/gdextension.cpp [hunk 1]
```diff
@@ -525,8 +513,7 @@ void GDExtension::deinitialize_library(InitializationLevel p_level) {
 }
 
 void GDExtension::_bind_methods() {
-	ClassDB::bind_method(D_METHOD("open_library", "path", "entry_symbol", "use_legacy_interface"), &GDExtension::open_library, DEFVAL(false));
-	ClassDB::bind_compatibility_method(D_METHOD("open_library", "path", "entry_symbol"), &GDExtension::open_library_compat_76406);
+	ClassDB::bind_method(D_METHOD("open_library", "path", "entry_symbol"), &GDExtension::open_library, DEFVAL(false));
```
Comment (RedworkDE) [inline core/extension/gdextension.cpp:516]:
```suggestion
	ClassDB::bind_method(D_METHOD("open_library", "path", "entry_symbol"), &GDExtension::open_library);
```

Comment (dsnopek) [reply core/extension/gdextension.cpp:516]: Thanks! Fixed in my latest push

**Review by mhilbrunner — APPROVED** [review]
Very much agree with the change. The code looks good to me, but TIWAGOS as most of this is gibberish to me :P

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**